ui/dashboards/src/context/DashboardProvider/DashboardProvider.tsx

```diff
@@ -60,14 +60,25 @@ export function useDashboardStore<T>(selector: (state: DashboardStoreState) => T
 }
 
 export function DashboardProvider(props: DashboardProviderProps) {
+  const [store] = useState(createDashboardStore(props)); // prevent calling createDashboardStore every time it rerenders
```

Comment:
Yay! 👏 

One small suggestion: use the callback form here to avoid `createDashboardStore` getting run every time this renders.

Reply:
Oh, also: what if the prop values change? Right now it seems like we'd just ignore it?

Reply:
> Oh, also: what if the prop values change?

hmm how about using a `key` for now to remount DashboardProvider?